lib/compress.js

```diff
@@ -179,16 +179,15 @@ merge(Compressor.prototype, {
     // we shouldn't compress (1,func)(something) to
     // func(something) because that changes the meaning of
     // the func (becomes lexical instead of global).
-    function maintain_call_binding(parent, orig, val) {
-        if (parent instanceof AST_Call && parent.expression === orig) {
-            if (val instanceof AST_PropAccess || (val instanceof AST_Symbol && val.name === "eval")) {
-                return make_node(AST_Seq, orig, {
-                    car: make_node(AST_Number, orig, {
-                        value: 0
-                    }),
-                    cdr: val
-                });
-            }
+    function maintain_this_binding(scope, parent, orig, val) {
+        if (parent instanceof AST_Call && parent.expression === orig && (val instanceof AST_PropAccess
+            || val instanceof AST_SymbolRef && val.name === "eval" && !scope.find_variable("eval"))) {
```

Comment:
You're right to doubt it, but that is in fact different. See [12.3.4.1 step 4](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-function-calls-runtime-semantics-evaluation). JavaScript is crazy.